static/js/snap-metrics.js

```diff
@@ -0,0 +1,285 @@
+     */
+    function formatXAxisTickLabels(x) {
+      return moment(x).format('MMM D');
+    }
+
+    /**
+     * Format the value displayed for each tick:
+     * - 10
+     * - 1.0k
+     * - 1.0m
+     * @param {number} y Value of the tick
+     */
+    function formatYAxisTickLabels(y) {
+      var str = y;
+      if (y >= 1000000) {
+        str = (y / 1000000).toFixed(1) + 'm';
+      } else if (y >= 1000) {
+        str = (y / 1000).toFixed(1) + 'k';
+      }
+      return str;
+    }
+
+    /**
+     * Formats axis and shows the graph.
+     */
+    function showGraphs() {
+      formatAxis(installsMetricsEl);
+      installsMetricsEl.style.opacity = 1;
+    }
+
+    var days = metrics.installs.buckets;
+    // Convert to moment object.
+    days = days.map(function (day) {
+      return moment(day);
+    });
+    // Prepend 'x'.
+    days.unshift('x'); 
+
+
+    var installs = metrics.installs.series[0].values;
+    // Prepend 'installs'.
+    installs.unshift(metrics.installs.series[0].name);
+
+    var mousePosition = {x: 0, y: 0};
+
+    // Installs Metrics
+    var installsMetricsEl = document.getElementById('installs_metrics');
+    var installsMetricsOffset = {
+      left: installsMetricsEl.offsetLeft,
+      top: installsMetricsEl.offsetTop
+    };
+    var installsMetrics = bb.generate({
+      bindto: '#installs_metrics',
+      legend: {
+        hide: true
+      },
+      padding: {
+        top: 0,
+        left: 72,
+        bottom: 0,
+        right: 112
+      },
+      tooltip: {
+        contents: snapcraftGraphTooltip,
+        position: positionTooltip
+      },
+      transition: {
+        duration: 0
+      },
+      point: {
+        focus: false
+      },
+      axis: {
+        x: {
+          tick: {
+            culling: false,
+            outer: true,
+            format: formatXAxisTickLabels
+          }
+        },
+        y: {
+          tick: {
+            format: formatYAxisTickLabels
+          }
+        }
+      },
+      bar: {
+        width: 4
+      },
+      resize: {
+        auto: false
+      },
+      data: {
+        colors: COLORS,
+        type: 'bar',
+        x: 'x',
+        columns: [
+          days,
+          installs
+        ]
+      }
+    });   
+
+    showGraphs();
+
+    // Extra events
+    var installMetricsElWidth = installsMetricsEl.clientWidth;
+    var resize = debounce(function () {
+      var el = installsMetricsEl;
+
+      if (el.clientWidth !== installMetricsElWidth) {
+        el.style.opacity = 0;
+        debounce(function () {
+          installsMetrics.resize();
+          showGraphs();
+          installMetricsElWidth = el.clientWidth;
+        }, 100)();
+      }
+    }, 500);
+
+    var mouseMove = function(e) {
+      mousePosition = {
+        x: e.x,
+        y: e.y
+      };
+    };
+
+    window.addEventListener('resize', resize);
+    window.addEventListener('mousemove', mouseMove);
+  }
+
+  window.renderMetrics = renderMetrics;
```

Comment:
We also need to think of how to do JS moving forward (meaning having JS functionality and data from the template). Exposing functions directly in global scope will not scale very well.

We could at least namespace it in `snapcraft.renderMetrics` or something. But also this feels so not like JS these days...

Thoughts?

Reply:
@bartaz I agree to all points. My only extra thought is that snapcraft.io will, for now, not have a lot of JS that is required for all pages and should be quite localised, so shouldn't be a massive issue.... For now.

I quite like `snapcraft.*` where we perhaps namespace `snapcraft.metrics.*`. But like you say, this doesn't feel hipster enough.

While we're thinking about this we should consider minification... Adding some cheeky ES[latest] and transpilation will sort this right out though AND we'll feel like the hippest hipsters on JavaScript Lane in Hipsterville.